syft/pkg/dotnet_deps_metadata.go

```diff
@@ -0,0 +1,24 @@
+package pkg
+
+import (
+	"github.com/anchore/packageurl-go"
+	"github.com/anchore/syft/syft/linux"
+)
+
+type DotnetDepsMetadata struct {
```

Comment:
This is probably a good opportunity to raise up the additional raw information for each of the packages in the `*.deps.json`, such as:
```
      "type": "package",
      "serviceable": true,
      "sha512": "sha512-kHBB+QmosVaG6DpngXQ8OlLVVNMzltNITfsRr68Z90qO7dSqJ2EHNd8dtBU1u3AQQLqqFHOY0lfmbpexeH6Pew==",
      "path": "awssdk.core/3.7.10.6",
      "hashPath": "awssdk.core.3.7.10.6.nupkg.sha512"
```

Straight from https://github.com/dotnet/cli/blob/v2.0.0/Documentation/specs/runtime-configuration-file.md

> The type property of a dependency object defines what kind of entity satisfied the dependency. Possible values include project and package (further comments on dependency types below).
...Open Question: type is also present in the libraries section. We don't really need it in both. It's in both now because the lock file does that and we want the formats to be similar. Should we remove it?

> # libraries Section (.deps.json)
This section contains a union of all the dependencies found in the various targets, and contains common metadata for them. Specifically, it contains the type, as well as a boolean indicating if the library can be serviced (serviceable, only for package-typed libraries) and a SHA-512 hash of the package file (sha512, only for package-typed libraries.

From a high level it seems like `path`, `sha512`, `hashPath` could be valuable to capture. I'm not certain of the value of `serviceable` or `type`, but wouldn't be against including these fields in `DotnetDepsMetadata`.

Reply:
This sounds good. I can include them. Not sure about `serviceable` and `type` too. Type would always be `package` (there's an if-statement) and `serviceable` will then always be true...

Reply:
Added `path`, `sha512` and `hashPath`